### PR TITLE
normed epigraph distance

### PR DESCRIPTION
**Pull request by matbesancon**
whenever a set is described as `fi(x) <= 0 for i in I`, defined the distance as a norm of the individual epigraph violations.

Many distance implementations are effectively following this guideline

### CODE REVIEW

File: src/distances.jl [hunk 1]
```diff
@@ -34,3 +26,23 @@ with `T` of appropriate type for members of the set.
...
+
+Distance used when `v ∈ s` can be expressed as a single scalar function `f(v) ≤ 0`.
+The distance is expressed as `max(f(v), 0)`.
+"""
+struct EpigraphViolationDistance <: AbstractDistance end
+
+"""
+    NormedEpigraphDistance{p}
+
+Distance used when `v ∈ s` can be expressed as a several functions `f_i(v) ≤ 0 ∀ i ∈ I`.
```
Comment (frapac) [inline src/distances.jl:41]: I would maybe rephrase, e.g. `as several functions` , or `as an intersection of different level sets`

File: src/distances.jl [hunk 2]
```diff
@@ -34,3 +26,23 @@ with `T` of appropriate type for members of the set.
...
+The distance is expressed as `norm([max(f_i(v), 0) ∀ i ∈ I], n)` with `p` a norm value (2, 1, Inf).
```
Comment (frapac) [inline src/distances.jl:42]: Could we use another norm value, apart of the usual ones `(2, 1, Inf)`?

File: src/distances.jl [hunk 3]
```diff
@@ -34,3 +26,23 @@ with `T` of appropriate type for members of the set.
...
+"""
+struct NormedEpigraphDistance{p} <: AbstractDistance end
+
+# _norm_set(::NormedEpigraphDistance{1}) = MOI.NormOneCone
```
Comment (frapac) [inline src/distances.jl:46]: Are these comments expected?

File: src/sets.jl [hunk 4]
```diff
@@ -124,31 +149,41 @@ function distance_to_set(::DefaultDistance, v::AbstractVector{<:Real}, s::MOI.Po
...
     result = s - u
-    return LinearAlgebra.norm2(push!(
-        max.(v[2:end], zero(result)),
-        max(result, zero(result)),
-    ))
+    return LinearAlgebra.norm(
+        push!(
+            max.(v[2:end], zero(result)),
+            max(result, zero(result)),
+        ), p,
```
Comment (frapac) [inline src/sets.jl:181]: Is the last comma expected?

File: src/distances.jl (hunk 2 above)
Comment (matbesancon) [reply src/distances.jl:42]: yes indeed, I rephrased it to indicate that any norm value that's legit for `LinearAlgebra.norm` can be accepted

File: src/distances.jl (hunk 1 above)
Comment (matbesancon) [reply src/distances.jl:41]: yes thanks, intersection of level sets is the best way to describe it IMO

File: src/distances.jl (hunk 3 above)
Comment (matbesancon) [reply src/distances.jl:46]: nope, removed, I thought I would have needed these, in the end nope

File: src/sets.jl (hunk 4 above)
Comment (matbesancon) [reply src/sets.jl:181]: yes, I try to leave trailing commas to be able to add a new line without changing the existing lines with arguments, maybe we could have the `p,` on a new line?
